pallets/proofs-dealer/src/benchmarking.rs

```diff
@@ -45,9 +83,192 @@ mod benchmarks {
         Ok(())
     }
 
+    #[benchmark]
+    fn submit_proof_no_checkpoint_challenges_key_proofs(
+        n: Linear<1, { T::MaxCustomChallengesPerBlock::get() }>,
+    ) -> Result<(), BenchmarkError> {
+        let n: u32 = n.into();
+        let (caller, proof) = setup_submit_proof::<T>(n)?;
+
+        // Call some extrinsic.
+        #[extrinsic_call]
+        Pallet::submit_proof(RawOrigin::Signed(caller.clone()), proof, None);
+
+        Ok(())
```

Comment:
Just as an integrity check I would add asserting that the last event was the `ProofAccepted` event for this provider

Reply:
+1